### DISCUSSION TITLE
[Feature]: 几个功能建议

### DISCUSSION THREAD

**Discussion by appotry** [walinejs/waline#1299] [Ideas]
### 功能概述 | Describe the feature

这个leancloud valine评论管理有几个很好的功能
https://github.com/HCLonely/Valine-Admin.git

![](https://images.weserv.nl?url=https://raw.githubusercontent.com/appotry/cloudimg/main/data/2022/08/21/20220821113832.webp)

如上图所示

- 是否邮件提醒 ，重新邮件提醒
- 是否微信提醒，重新微信提醒
- 查看评论是快捷跳转到指定评论，而不是跳转到评论框头部

个人发现 valine waline 评论id生成方式都是一样的, 网页中加入如下代码，就可以评论ID跳转到指定评论

```html
  // 评论跳转
  if(window.location.hash){
    var checkExist = setInterval(function() {
        if ($(window.location.hash).length) {
            $('html, body').animate({scrollTop: $(window.location.hash).offset().top-90}, 1000);
            clearInterval(checkExist);
        }
    }, 100);
  }
```

**Comment by lizheming** [comment]
1. 决定一个邮件通知是否真实通知到的条件有很多，只是发送成功了意义并不大，包括是否被对方邮件系统防火墙拦截，是否进入垃圾邮件等都是因素。所以后台查看邮件是否已通知的意义其实不是特别大，你可以认为只要你配置了邮件通知系统那一定是已经发送了通知的。Valine-Admin 有这个标记的原因是因为它是定时任务运行的原因，而 Waline 是发送时及时通知的，根本不需要这么麻烦。
2. 微信提醒状态查看这个意义就更不大了。首先微信提醒只能提醒博主本人，而如果你已经在后台看到了消息，那我也不需要关心通知是不是发送过去了。毕竟通知的本质就是让人知道有消息了。
3. 锚点这个是所有动态加载评论系统的痛点，但你提供的方案是无法完美解决这个问题的。一是你这个代码会造成死循环，二是方案是无法实现锚点到下一页之后的评论的，仅限第一页的评论。

**Reply by appotry** [reply]
有道理，那么这几个建议算无效了

有个问题，个人同时使用Valine-Admin  和waline 后台管理评论，但从来只看到 Valine-Admin 发送的邮件通知，这是从 Valine-Admin 邮件模板美化判断出来的

waline 后台检查邮件通知的评率是多少，怎么判断的了？

 Valine-Admin 发送了邮件通知之后，waline就不发送了？

**Reply by appotry** [reply]
不过，确认waline发送邮件成功还是有意义的，对方防火墙或者垃圾拦截是另外一回事

记得邮件发送有个发送成功回执，不少邮箱服务商提供这个功能，这个waline是否能实现

**Reply by lizheming** [reply]
两个系统没有什么关联性，不要拿 Valine-Admin 的逻辑去思考 Waline，我已经说过 Waline 是在发送评论的时候发送通知的，何来频率一说？至于回执，那是邮箱提供的功能，Waline 后台暂不考虑这个特性
